rio/srcrobot/subsystems/Shooter.py

```diff
@@ -38,21 +38,36 @@ def __init__(self):
         self.VelocityControl = VelocityVoltage(0)
         self.VoltageControl = VoltageOut(0)
         self.percentOutput = DutyCycleOut(0)
-        
+
+        self.currentShotVelocity = 0.0
+
+        self.topShooterVelocitySupplier = self.topShooter.get_velocity().as_supplier()
+        self.bottomShooterVelocitySupplier = self.bottomShooter.get_velocity().as_supplier()
+
+    def setShooterVelocity(self, velocity):
+        self.topShooter.set_control(self.VelocityControl.with_velocity(Conversions.MPSToRPS(velocity,  0.101 * math.pi)))
+        self.bottomShooter.set_control(self.VelocityControl.with_velocity(Conversions.MPSToRPS(velocity,  0.101 * math.pi)))
+
+        self.currentShotVelocity = Conversions.MPSToRPS(velocity,  0.101 * math.pi)
+
+    def neutralizeShooter(self):
+        self.topShooter.set_control(self.VoltageControl.with_output(0))
+        self.bottomShooter.set_control(self.VoltageControl.with_output(0))
+
+    def idle(self):
+        return self.run(self.setShooterVelocity(Constants.ShooterConstants.kAmpShootSpeed))    
+    
     def shoot(self):
-        # return InstantCommand(lambda: self.bottomShooter.set_control(self.VelocityControl.with_velocity(Conversions.MPSToRPS(Constants.ShooterConstants.kPodiumShootSpeed,  0.101 * math.pi)))).alongWith(
-        #     InstantCommand(lambda:self.topShooter.set_control(self.VelocityControl.with_velocity(Conversions.MPSToRPS(Constants.ShooterConstants.kPodiumShootSpeed,  0.101 * math.pi))))
-        # )
-        return InstantCommand(lambda: self.bottomShooter.set_control(self.VelocityControl.with_velocity(Conversions.MPSToRPS(Constants.ShooterConstants.kPodiumShootSpeed,  0.101 * math.pi)))).alongWith(
-            InstantCommand(lambda:self.topShooter.set_control(self.VelocityControl.with_velocity(Conversions.MPSToRPS(Constants.ShooterConstants.kPodiumShootSpeed,  0.101 * math.pi))))
-        )
+        return self.run(lambda: self.setShooterVelocity(Constants.ShooterConstants.kPodiumShootSpeed))
 
     def shootReverse(self):
-        return InstantCommand(lambda: self.bottomShooter.set_control(self.VelocityControl.with_velocity(Conversions.MPSToRPS(-Constants.ShooterConstants.kPodiumShootSpeed,  0.101 * math.pi)))).alongWith(
-            InstantCommand(lambda:self.topShooter.set_control(self.VelocityControl.with_velocity(Conversions.MPSToRPS(-Constants.ShooterConstants.kPodiumShootSpeed,  0.101 * math.pi))))
-        )
+        return self.run(lambda: self.setShooterVelocity(-Constants.ShooterConstants.kPodiumShootSpeed))
+    
+    def isShooterReady(self):
+        topShooterReady = abs(abs(self.topShooterVelocitySupplier()) - self.currentShotVelocity) < 0.5
+        bottomShooterReady = abs(abs(self.bottomShooterVelocitySupplier()) - self.currentShotVelocity) < 0.5
```

Comment:
I should have caught this the first time but the inner abs() should not be in either of these conditions. You want the difference between the current velocity and the desired velocity, not the difference between the absolute value of the current velocity and the desired velocity. So, something like this:
```
topShooterReady = abs(self.topShooterVelocitySupplier() - self.currentShotVelocity) < 0.5
bottomShooterReady = abs(self.bottomShooterVelocitySupplier() - self.currentShotVelocity) < 0.5
```